### PR TITLE
Feat(velaql): read max depth for query iterator from env.

### PR DESCRIPTION
**Pull request by qashlan**
Ref #6014

### Description of your changes

Read max depth from env if not found return the default const which is 5

### CODE REVIEW

File: pkg/velaql/providers/query/tree.go [hunk 1]
```diff
@@ -48,11 +50,23 @@ import (
...
+)
+
 // relationshipKey is the configmap key of relationShip rule
 var relationshipKey = "rules"
 
-// set the iterator max depth is 5
-var maxDepth = 5
+// check if max depth is provided or return the default max depth
+var maxDepth = func() int {
+	maxDepth, err := strconv.Atoi(os.Getenv("KUBEVELA_QUERYTREE_MAXDEPTH"))
```
[the remark this answers is not in the document]
Comment (qashlan) [reply pkg/velaql/providers/query/tree.go:63]: @StevenLeiZhang yes I thought about it and even asked if I should place it in the docs or something, and I still have trouble finding where to put the value , I know its somewhere in the charts folder any info/help would be really appreciated

[the remark this answers is not in the document]
Comment (Somefive) [reply pkg/velaql/providers/query/tree.go:63]: KUBEVELA_QUERYTREE_MAXDEPTH is an environment variable and it is a client-side setting (used in vela cli). So I think it does not need to be configured in the chart.

File: pkg/velaql/providers/query/tree.go [hunk 2]
```diff
@@ -48,11 +50,23 @@ import (
...
+	if err != nil {
```
Comment (Somefive) [inline pkg/velaql/providers/query/tree.go:64]: if err != nil || maxDepth <= 0

File: pkg/velaql/providers/query/tree.go [hunk 3]
```diff
@@ -48,11 +50,23 @@ import (
...
+		return DefaultMaxDepth
+	}
+
+	return maxDepth
+}()
```
Comment (Somefive) [inline pkg/velaql/providers/query/tree.go:69]: You can leave this function to be called where maxDepth is used.

**Review by Somefive — APPROVED** [review]

**Pull request merged**